easybuild/easyblocks/m/mesa.py

```diff
@@ -0,0 +1,122 @@
+                }
+                # determine list of values to pass to swr-arches configuration option
+                cpu_features = get_cpu_features()
+                self.swr_arches = sorted([swrarch for feat, swrarch in feat_to_swrarch.items() if feat in cpu_features])
+
+    def configure_step(self):
+        """
+        Customise the configure options based on the processor architecture of the host
+        (x86_64 or not, CPU features, ...)
+        """
+
+        arch = get_cpu_architecture()
+        if 'gallium-drivers' not in self.cfg['configopts']:
+            # Install appropriate Gallium drivers for current architecture
+            if arch == X86_64:
+                self.cfg.update('configopts', "-Dgallium-drivers='swrast,swr'")
+            elif arch == POWER:
+                self.cfg.update('configopts', "-Dgallium-drivers='swrast'")
+
+        if self.swr_arches:
+            self.cfg.update('configopts', '-Dswr-arches=' + ','.join(self.swr_arches))
+
+        return super(EB_Mesa, self).configure_step()
+
+    def install_step(self):
+        """Also copy additional header files after installing Mesa."""
+
+        super(EB_Mesa, self).install_step()
+
+        # also install header files located in include/GL/internal, unless they're available already;
+        # we can't enable both DRI and Gallium drivers,
+        # but we can provide the DRI header file (GL/internal/dri_interface.h)
+        target_inc_GL_internal = os.path.join(self.installdir, 'include', 'GL', 'internal')
+        if not os.path.exists(target_inc_GL_internal):
+            src_inc_GL_internal = os.path.join(self.start_dir, 'include', 'GL', 'internal')
+            copy_dir(src_inc_GL_internal, target_inc_GL_internal)
+            self.log.info("Copied %s to %s" % (src_inc_GL_internal, target_inc_GL_internal))
+
+    def sanity_check_step(self):
+        """Custom sanity check for Mesa."""
+
+        shlib_ext = get_shared_lib_ext()
+
+        if LooseVersion(self.version) >= LooseVersion('20.0'):
+            header_files = [os.path.join('include', 'EGL', x) for x in ['eglmesaext.h', 'eglextchromium.h']]
+            header_files.extend([
+                os.path.join('include', 'GL', 'osmesa.h'),
+                os.path.join('include', 'GL', 'internal', 'dri_interface.h'),
+            ])
+        else:
+            gl_inc_files = ['glext.h', 'gl_mangle.h', 'glx.h', 'osmesa.h', 'gl.h', 'glxext.h', 'glx_mangle.h']
+            gles_inc_files = [('GLES', 'gl.h'), ('GLES2', 'gl2.h'), ('GLES3', 'gl3.h')]
+            header_files = [os.path.join('include', 'GL', x) for x in gl_inc_files]
+            header_files.extend([os.path.join('include', x, y) for (x, y) in gles_inc_files])
+
+        custom_paths = {
+            'files': [os.path.join('lib', 'libOSMesa.%s' % shlib_ext)] + header_files,
+            'dirs': [os.path.join('include', 'GL', 'internal')],
+        }
+
+        if self.swr_arches:
```

Comment:
Whoops, this is indeed wrong: It expects swr to be build without checking the configopts

Reply:
@Flamefire Did this get fixed?

Reply:
I guess it was in #2006...